src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int GROW_COEFFICIENT = 2;
+    private int currentCapacity;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            createDefaultTable();
+        }
+        if (size == threshold) {
+            resize();
+        }
+        int index = getIndex(hash(key));
+        putNewNodeInTheTable(index, key, value);
     }
 
     @Override
     public V getValue(K key) {
+        if (currentCapacity == 0) {
+            return null;
+        }
```

Comment:
```suggestion
```

Reply:
If I delete this, I can have ArithmeticException: / by zero .  It happens when you didn't put something in your HashMap (currentCapacity = 0) but called the get method. key.hashCode() % currentCapacity